src/main/scala/com/gu/subscriptions/cas/directives/ProxyDirective.scala

```diff
@@ -134,28 +134,67 @@ trait ProxyDirective extends Directives with ErrorRoute with LazyLogging {
         // return error: "Mandatory data missing from request"
   }
 
-  def zuoraRoute(subsReq: SubscriptionRequest): Route = zuoraDirective(subsReq) { (activation, subscriptionName) =>
-    val validSubscription = subscriptionService.getValidSubscription(Name(subscriptionName.get.trim.dropWhile(_ == '0')), subsReq.password.getOrElse(""))
 
-    validSubscription.onFailure {
-      case t: Throwable =>
-        logger.error(s"Failed getting Zuora subscription ${t.getMessage} ${subsReq.subscriberId}")
-        throw t
+
+  def processEmergencyToken(req: SubscriptionRequest):Option[SubscriptionExpiration] = {
+    import com.gu.subscriptions.cas.model.TokenPayloadOps._
+
+    req.subscriberId.flatMap{ rawSubscriberId1 =>
+      val subsId = rawSubscriberId1.trim.toUpperCase
+
+      def processPayload(payloadResult: PayloadResult) = payloadResult match {
+        case Valid(payload) =>
+          logger.info(s"subscriber id:'$subsId' was created on ${payload.creationDate}")
+
+          val subExp = SubscriptionExpiration(
+            expiryDate = payload.expiryDate,
+            expiryType = ExpiryType.SUB,
+            subscriptionCode = Some(payload.subscriptionCode),
+            provider = Some(Configuration.EmergencyTokens.prefix)
+          )
+          Some(subExp)
+        case _ => None
+      }
+
+    if (!subsId.startsWith(emergencyTokens.prefix)) {
+      None
+    } else {
+      logger.info(s"EMERGENCY PROVIDER triggered for subscriber id:'$subsId' password: '${req.password} deviceId:${req.deviceId}")
+      val payloadResult = emergencyTokens.decoder.decode(subsId) //todo for some reason this throws an exception with an invalid token...
```

Comment:
It looks like this only happens if a valid emergency token with some at least one additional character appended at the end is provided in the call.
This seems to also happen in the current master version